### ISSUE TITLE
Feature request: Option to exclude the latest version of each artifact from Cleanup Policies

### ISSUE THREAD

**Issue by beppo-sturmtief** [sonatype/nexus-public#400]
In NR OSS, Cleanup Policies are limited to filtering a) name based, b) based on artifact age and c) based on activity (age of last download of the artifact).

This leads into problems with stable artifacts - if one is not rebuild & deployed (and is cached in all local maven repos) for longer than the cleanup policy, the artifact gets deleted and is no longer available from the repo.

This issue is actively preventing me from employing automatic cleanup policies, thereby effectively keeping my repo overgrown and cluttered with outdated snapshot artifacts.

It was reported that a comprehensive feature to adress this in the works for NR PRO, but OSS users are excluded. Also notably, there was a feature in the free version of Nexus 2.

To relieve the issue for OSS users, i propose a minimal solution that also does not compete with the PRO feature: add a boolean option to the cleanup option to never remove the latest version   of any artifact selected by the filter settings.

This would imo be a feature of limited scope, but would address the issue of stable artifacts for most non-power users.

See also issue #246 and others.

 Would be greatly appreciated!

**Comment by nblair** [comment]
Thank you @beppo-sturmtief - we'll keep this enhancement idea open to track demand.

**Comment by QuimaxW** [comment]
For a Nuget respository, this would be a huge life-saver for me. Ideally a "last x versions" would be perfect, but I'd accept the option to not delete the single last artifact that's left.

**Comment by srmo** [comment]
Any updates on this?

**Comment by cvmocanu** [comment]
First, thanks for Nexus, and making a community edition available for free.

I just wanted to say that without this option, the "cleanup policies" are unusable for me. I can't risk not being able to build the product because all versions of the common-lib library were deleted (if I didn't release more often than the cutoff time).

**Comment by Surfjoan** [comment]
I agree.. I currently manage a Nexux server with over 4 TB of docker images.  I really would like to to use cleanup policies more aggressivly but i cannot risk removing old components still in use. Therefore i need an option to keep at least the latest version of an artifact, but even better, to keep the last X artifacts. With google i have found people fixing this "glitch" by scripts but i cannot understand why it is not build in
